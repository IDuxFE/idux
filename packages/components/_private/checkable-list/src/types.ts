/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { VirtualScrollToFn } from '@idux/cdk/scroll'
import type { ExtractInnerPropTypes, ExtractPublicPropTypes, MaybeArray, VKey } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes, PropType } from 'vue'

export interface CheckableListData {
  key?: VKey
  label?: string
  disabled?: boolean
  [key: string]: unknown
}

export interface CheckableListScroll {
  height?: string | number
  fullHeight?: boolean
}

export const checkableListProps = {
  dataSource: Array as PropType<CheckableListData[]>,
  checkable: {
    type: Boolean,
    default: true,
  },
  removable: {
    type: Boolean,
    default: false,
  },
  checked: Function as PropType<(item: CheckableListData) => boolean>,
  customAdditional: { type: Function as PropType<CheckableListCustomAdditional>, default: undefined },
  disabled: Function as PropType<(item: CheckableListData) => boolean>,
  getKey: Function as PropType<(item: CheckableListData) => VKey>,
  labelKey: String,
  virtual: {
    type: Boolean,
    default: false,
  },
  scroll: Object as PropType<CheckableListScroll>,
  onCheckChange: [Function, Array] as PropType<MaybeArray<(item: CheckableListData, checked: boolean) => void>>,
  onRemove: [Function, Array] as PropType<MaybeArray<(item: CheckableListData) => void>>,
  onScroll: [Function, Array] as PropType<MaybeArray<(evt: Event) => void>>,
  onScrolledChange: [Function, Array] as PropType<
    MaybeArray<(startIndex: number, endIndex: number, visibleData: unknown[]) => void>
  >,
  onScrolledBottom: [Function, Array] as PropType<MaybeArray<() => void>>,
} as const

export const checkableListItemProps = {
  checked: {
    type: Boolean,
    default: false,
  },
  checkable: {
    type: Boolean,
    default: true,
  },
  removable: {
    type: Boolean,
    default: false,
  },
  disabled: {
    type: Boolean,
    default: false,
  },
  label: String,
  value: {
    type: [String, Number, Symbol] as PropType<VKey>,
    required: true,
  },
  onCheckChange: [Function, Array] as PropType<MaybeArray<(checked: boolean) => void>>,
  onRemove: [Function, Array] as PropType<MaybeArray<() => void>>,
} as const

export interface CheckableListApi {
  scrollTo: VirtualScrollToFn
}

export type CheckableListProps = ExtractInnerPropTypes<typeof checkableListProps>
export type CheckableListItemProps = ExtractInnerPropTypes<typeof checkableListItemProps>
export type CheckableListPublicProps = ExtractPublicPropTypes<typeof checkableListProps>
export type CheckableListComponent = DefineComponent<
  Omit<HTMLAttributes, keyof CheckableListPublicProps> & CheckableListPublicProps,
  CheckableListApi
>
export type CheckableListInstance = InstanceType<DefineComponent<CheckableListProps, CheckableListApi>>

export type CheckableListCustomAdditional = (options: {
  data: CheckableListData
  index: number
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
}) => Record<string, any> | undefined
