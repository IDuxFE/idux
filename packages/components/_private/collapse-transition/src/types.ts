/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes } from 'vue'

import { IxPropTypes } from '@idux/cdk/utils'

export type CollapseTransitionMode = 'height' | 'width'

export const collapseTransitionProps = {
  appear: IxPropTypes.bool.def(false),
  name: IxPropTypes.string,
  mode: IxPropTypes.oneOf<CollapseTransitionMode>(['height', 'width']).def('height'),
  onAfterEnter: IxPropTypes.emit(),
  onAfterLeave: IxPropTypes.emit(),
}

export type CollapseTransitionProps = ExtractInnerPropTypes<typeof collapseTransitionProps>
export type CollapseTransitionPublicProps = ExtractPublicPropTypes<typeof collapseTransitionProps>
export type CollapseTransitionComponent = DefineComponent<
  Omit<HTMLAttributes, keyof CollapseTransitionPublicProps> & CollapseTransitionPublicProps
>
export type CollapseTransitionInstance = InstanceType<DefineComponent<CollapseTransitionProps>>
