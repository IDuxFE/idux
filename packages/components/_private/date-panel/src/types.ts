/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes, MaybeArray } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes, PropType } from 'vue'

export const datePanelProps = {
  cellTooltip: Function as PropType<(cell: { value: Date; disabled: boolean }) => string | void>,
  disabledDate: Function as PropType<(date: Date) => boolean>,
  type: {
    type: String as PropType<DatePanelType>,
    default: 'date',
  },
  value: [Date, Array] as PropType<Date | (Date | undefined)[]>,
  activeDate: Date,
  visible: {
    type: Boolean,
    default: undefined,
  },
  isSelecting: {
    type: Boolean,
    default: undefined,
  },

  onCellClick: [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
  onCellMouseenter: [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
  'onUpdate:activeDate': [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
}

export type DatePanelProps = ExtractInnerPropTypes<typeof datePanelProps>
export type DatePanelPublicProps = ExtractPublicPropTypes<typeof datePanelProps>
export type DatePanelComponent = DefineComponent<
  Omit<HTMLAttributes, keyof DatePanelPublicProps> & DatePanelPublicProps
>
export type DatePanelInstance = InstanceType<DefineComponent<DatePanelProps>>

export type DatePanelType = 'date' | 'week' | 'month' | 'quarter' | 'year'

// private
export const panelRowProps = {
  rowIndex: {
    type: Number,
    required: true,
  },
}

export const panelCellProps = {
  rowIndex: {
    type: Number,
    required: true,
  },
  cellIndex: {
    type: Number,
    required: true,
  },
}
