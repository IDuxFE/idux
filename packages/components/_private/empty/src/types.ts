/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { EmptyProps as IxEmptyProps } from '@idux/components/empty'

export interface EmptyProps {
  empty?: string | IxEmptyProps
}
