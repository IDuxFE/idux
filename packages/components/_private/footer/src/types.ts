/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes, VKey } from '@idux/cdk/utils'
import type { ButtonProps } from '@idux/components/button'
import type { DefineComponent, HTMLAttributes, PropType, VNode } from 'vue'

export interface FooterButtonProps extends ButtonProps {
  key?: VKey
  text?: string | VNode
  onClick?: (evt: Event) => void
}

export const footerProps = {
  cancel: Function as PropType<(evt?: Event | unknown) => Promise<void> | void>,
  cancelButton: Object as PropType<ButtonProps>,
  cancelLoading: Boolean,
  cancelText: String,
  cancelVisible: {
    type: Boolean,
    default: true,
  },
  footer: [Boolean, Array, Object] as PropType<boolean | FooterButtonProps[] | VNode>,
  ok: Function as PropType<(evt?: Event | unknown) => Promise<void> | void>,
  okButton: Object as PropType<ButtonProps>,
  okLoading: Boolean,
  okText: String,
} as const

export type FooterProps = ExtractInnerPropTypes<typeof footerProps>
export type FooterPublicProps = ExtractPublicPropTypes<typeof footerProps>
export type FooterComponent = DefineComponent<Omit<HTMLAttributes, keyof FooterPublicProps> & FooterPublicProps>
export type FooterInstance = InstanceType<DefineComponent<FooterProps>>
