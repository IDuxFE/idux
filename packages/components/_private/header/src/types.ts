/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { HeaderProps as IxHeaderProps } from '@idux/components/header'
import type { VNode } from 'vue'

export interface HeaderProps {
  closable?: boolean
  closeIcon?: string | VNode
  header?: string | IxHeaderProps
  size?: 'lg' | 'md' | 'sm'
  onClose?: (evt: Event) => void
}
