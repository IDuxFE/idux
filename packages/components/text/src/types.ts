/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes } from '@idux/cdk/utils'
import type { TooltipProps } from '@idux/components/tooltip'
import type { Component, DefineComponent, HTMLAttributes, PropType } from 'vue'

export const textProps = {
  copyable: { type: Boolean, default: false },
  expandable: { type: Boolean, default: false },
  lineClamp: { type: [String, Number] as PropType<string | number>, default: undefined },
  tag: { type: [String, Object] as PropType<string | Component>, default: 'span' },
  tooltip: { type: [Boolean, String, Object] as PropType<boolean | 'native' | TooltipProps>, default: true },
} as const

export type TextProps = ExtractInnerPropTypes<typeof textProps>
export type TextPublicProps = ExtractPublicPropTypes<typeof textProps>
export type TextComponent = DefineComponent<Omit<HTMLAttributes, keyof TextPublicProps> & TextPublicProps>
export type TextInstance = InstanceType<DefineComponent<TextProps>>
