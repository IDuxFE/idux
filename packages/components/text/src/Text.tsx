/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import { type Slot, computed, defineComponent, normalizeClass, shallowRef } from 'vue'

import { isObject, isString, throttle } from 'lodash-es'

import { useClipboard } from '@idux/cdk/clipboard'
import { getFirstValidNode, useState } from '@idux/cdk/utils'
import { useGlobalConfig } from '@idux/components/config'
import { IxIcon } from '@idux/components/icon'
import { IxTooltip } from '@idux/components/tooltip'

import { textProps } from './types'

export default defineComponent({
  name: 'IxText',
  inheritAttrs: false,
  props: textProps,
  setup(props, { attrs, slots }) {
    const common = useGlobalConfig('common')
    const mergedPrefixCls = computed(() => `${common.prefixCls}-text`)

    const elementRef = shallowRef<HTMLElement>()
    const [mergedDisabled, setDisabled] = useState(true)
    const [mergedEntered, setEntered] = useState(false)
    const [mergedExpanded, setExpanded] = useState(false)
    const [mergedCopied, setCopied] = useState(false)

    const { copy } = useClipboard()

    const checkDisabled = () => {
      const element = elementRef.value
      if (mergedExpanded.value || !element) {
        return true
      }

      const { lineClamp } = props
      if (lineClamp !== undefined) {
        return element.scrollHeight <= element.offsetHeight
      } else {
        const currOverflow = element.style.overflow
        if (!currOverflow || currOverflow === 'visible') {
          element.style.overflow = 'hidden'
        }
        const isOverflowing = element.clientWidth < element.scrollWidth || element.clientHeight < element.scrollHeight
        element.style.overflow = currOverflow
        return !isOverflowing
      }
    }

    const onClick = () => {
      const nextExpanded = !mergedExpanded.value
      setExpanded(nextExpanded)
      setDisabled(nextExpanded)
    }

    const onMouseEnter = () => {
      setEntered(true)
      setDisabled(checkDisabled())
    }

    const onCopy = throttle((text: string) => {
      if (!text || mergedCopied.value) {
        return
      }
      copy(text).then(success => {
        if (success) {
          setCopied(true)
          setTimeout(() => setCopied(false), 3000)
        }
      })
    }, 300)

    const classes = computed(() => {
      const prefixCls = mergedPrefixCls.value
      const { lineClamp } = props
      return normalizeClass({
        [prefixCls]: true,
        [`${prefixCls}-line-clamp`]: lineClamp !== undefined,
      })
    })

    const style = computed(() => {
      const { expandable, lineClamp } = props
      const expanded = mergedExpanded.value
      return {
        cursor: expandable ? 'pointer' : '',
        'text-overflow': !expanded && lineClamp === undefined ? 'ellipsis' : '',
        '-webkit-line-clamp': !expanded && lineClamp !== undefined ? lineClamp : '',
      }
    })

    return () => {
      const { tag: Tag, tooltip, expandable, copyable } = props
      const prefixCls = mergedPrefixCls.value
      const disabled = mergedDisabled.value
      const entered = mergedEntered.value
      const isNative = tooltip === 'native'
      const titleSlot = slots.title || slots.default

      let node = (
        <Tag
          ref={elementRef}
          class={classes.value}
          style={style.value}
          title={isNative && !disabled ? getStringBySlot(titleSlot) : undefined}
          onClick={expandable ? onClick : undefined}
          onMouseenter={onMouseEnter}
          {...attrs}
        >
          {slots.default?.()}
        </Tag>
      )

      if (!disabled && entered && tooltip && !isNative) {
        const tooltipProps = isObject(tooltip) ? { ...tooltip, disabled } : { disabled }
        node = (
          <IxTooltip {...tooltipProps} v-slots={{ title: titleSlot }}>
            {node}
          </IxTooltip>
        )
      }

      if (copyable) {
        const copied = mergedCopied.value
        const copyIcon = slots.copyIcon ? slots.copyIcon({ copied }) : <IxIcon name={copied ? 'check' : 'copy'} />
        node = (
          <div class={`${prefixCls}-wrapper`}>
            {node}
            <span class={`${prefixCls}-copy-icon`} onClick={() => onCopy(getStringBySlot(slots.default))}>
              {copyIcon}
            </span>
          </div>
        )
      }

      return node
    }
  },
})

function getStringBySlot(slot: Slot | undefined) {
  const validNode = getFirstValidNode(slot?.())
  if (!validNode) {
    return ''
  }
  const { children } = validNode
  if (isString(children)) {
    return children
  }
  return ''
}
