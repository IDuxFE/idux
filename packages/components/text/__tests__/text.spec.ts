import { MountingOptions, mount } from '@vue/test-utils'

import { renderWork } from '@tests'

import Text from '../src/Text'
import { TextProps } from '../src/types'

const defaultSlot =
  () => `@idux 是一套企业级中后台 UI 组件库, 致力于提供高效愉悦的开发体验。 基于 Vue 3.x + TypeScript 开发,
      全部代码开源并遵循 MIT 协议，任何企业、组织及个人均可免费使用。`

describe('Text', () => {
  const TextMount = (options?: MountingOptions<Partial<TextProps>>) => {
    const { slots } = options || {}
    return mount(Text, { ...(options as MountingOptions<TextProps>), slots: { default: defaultSlot, ...slots } })
  }

  renderWork<TextProps>(Text, {
    slots: { default: defaultSlot },
  })

  test('copyable work', async () => {
    const wrapper = TextMount({ props: { copyable: true } })

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.classes()).toContain('ix-text-wrapper')

    document.execCommand = vi.fn().mockReturnValue(true)
    await wrapper.find('.ix-text-copy-icon').trigger('click')

    expect(document.execCommand).toHaveBeenCalledWith('copy')

    await wrapper.setProps({ copyable: false })

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.classes()).not.toContain('ix-text-wrapper')
    expect(wrapper.classes()).toContain('ix-text')
  })

  // 需要 E2E 测试
  test('expandable work', async () => {
    const wrapper = TextMount({ props: { lineClamp: 2, expandable: true } })

    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.trigger('click')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('lineClamp work', async () => {
    const wrapper = TextMount({ props: { lineClamp: 2 } })

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.classes()).toContain('ix-text-line-clamp')

    await wrapper.setProps({ lineClamp: undefined })

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.classes()).not.toContain('ix-text-line-clamp')
  })

  test('tag work', async () => {
    const wrapper = TextMount({ props: { tag: 'div' } })

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.element.tagName).toBe('DIV')

    await wrapper.setProps({ tag: undefined })

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.element.tagName).toBe('SPAN')
  })

  test('tooltip work', async () => {
    const wrapper = TextMount({ props: { tooltip: 'native' } })

    // 无效
    await wrapper.trigger('mouseenter')

    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.setProps({ tooltip: true })

    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.setProps({ tooltip: false })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
