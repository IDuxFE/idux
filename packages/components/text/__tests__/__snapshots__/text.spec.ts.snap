// Vitest Snapshot v1

exports[`Text > copyable work 1`] = `
"<div class=\\"ix-text-wrapper\\"><span class=\\"ix-text\\" style=\\"text-overflow: ellipsis;\\">@idux 是一套企业级中后台 UI 组件库, 致力于提供高效愉悦的开发体验。 基于 Vue 3.x + TypeScript 开发,
      全部代码开源并遵循 MIT 协议，任何企业、组织及个人均可免费使用。</span><span class=\\"ix-text-copy-icon\\"><i class=\\"ix-icon ix-icon-copy\\" role=\\"img\\" aria-label=\\"copy\\"></i></span></div>"
`;

exports[`Text > copyable work 2`] = `
"<span class=\\"ix-text\\" style=\\"text-overflow: ellipsis;\\">@idux 是一套企业级中后台 UI 组件库, 致力于提供高效愉悦的开发体验。 基于 Vue 3.x + TypeScript 开发,
      全部代码开源并遵循 MIT 协议，任何企业、组织及个人均可免费使用。</span>"
`;

exports[`Text > expandable work 1`] = `
"<span class=\\"ix-text ix-text-line-clamp\\" style=\\"cursor: pointer;\\">@idux 是一套企业级中后台 UI 组件库, 致力于提供高效愉悦的开发体验。 基于 Vue 3.x + TypeScript 开发,
      全部代码开源并遵循 MIT 协议，任何企业、组织及个人均可免费使用。</span>"
`;

exports[`Text > expandable work 2`] = `
"<span class=\\"ix-text ix-text-line-clamp\\" style=\\"cursor: pointer;\\">@idux 是一套企业级中后台 UI 组件库, 致力于提供高效愉悦的开发体验。 基于 Vue 3.x + TypeScript 开发,
      全部代码开源并遵循 MIT 协议，任何企业、组织及个人均可免费使用。</span>"
`;

exports[`Text > lineClamp work 1`] = `
"<span class=\\"ix-text ix-text-line-clamp\\">@idux 是一套企业级中后台 UI 组件库, 致力于提供高效愉悦的开发体验。 基于 Vue 3.x + TypeScript 开发,
      全部代码开源并遵循 MIT 协议，任何企业、组织及个人均可免费使用。</span>"
`;

exports[`Text > lineClamp work 2`] = `
"<span class=\\"ix-text\\" style=\\"text-overflow: ellipsis;\\">@idux 是一套企业级中后台 UI 组件库, 致力于提供高效愉悦的开发体验。 基于 Vue 3.x + TypeScript 开发,
      全部代码开源并遵循 MIT 协议，任何企业、组织及个人均可免费使用。</span>"
`;

exports[`Text > render work 1`] = `
"<span class=\\"ix-text\\" style=\\"text-overflow: ellipsis;\\">@idux 是一套企业级中后台 UI 组件库, 致力于提供高效愉悦的开发体验。 基于 Vue 3.x + TypeScript 开发,
      全部代码开源并遵循 MIT 协议，任何企业、组织及个人均可免费使用。</span>"
`;

exports[`Text > tag work 1`] = `
"<div class=\\"ix-text\\" style=\\"text-overflow: ellipsis;\\">@idux 是一套企业级中后台 UI 组件库, 致力于提供高效愉悦的开发体验。 基于 Vue 3.x + TypeScript 开发,
  全部代码开源并遵循 MIT 协议，任何企业、组织及个人均可免费使用。</div>"
`;

exports[`Text > tag work 2`] = `
"<span class=\\"ix-text\\" style=\\"text-overflow: ellipsis;\\">@idux 是一套企业级中后台 UI 组件库, 致力于提供高效愉悦的开发体验。 基于 Vue 3.x + TypeScript 开发,
      全部代码开源并遵循 MIT 协议，任何企业、组织及个人均可免费使用。</span>"
`;

exports[`Text > tooltip work 1`] = `
"<span class=\\"ix-text\\" style=\\"text-overflow: ellipsis;\\">@idux 是一套企业级中后台 UI 组件库, 致力于提供高效愉悦的开发体验。 基于 Vue 3.x + TypeScript 开发,
      全部代码开源并遵循 MIT 协议，任何企业、组织及个人均可免费使用。</span>"
`;

exports[`Text > tooltip work 2`] = `
"<span class=\\"ix-text\\" style=\\"text-overflow: ellipsis;\\">@idux 是一套企业级中后台 UI 组件库, 致力于提供高效愉悦的开发体验。 基于 Vue 3.x + TypeScript 开发,
      全部代码开源并遵循 MIT 协议，任何企业、组织及个人均可免费使用。</span>"
`;

exports[`Text > tooltip work 3`] = `
"<span class=\\"ix-text\\" style=\\"text-overflow: ellipsis;\\">@idux 是一套企业级中后台 UI 组件库, 致力于提供高效愉悦的开发体验。 基于 Vue 3.x + TypeScript 开发,
      全部代码开源并遵循 MIT 协议，任何企业、组织及个人均可免费使用。</span>"
`;
